tests/introspection/ApiTypoTest.cs

```diff
@@ -713,9 +714,39 @@ public void TypoTest ()
 
 					var methods = t.GetMethods ();
 					foreach (MethodInfo m in methods) {
-					if (!m.IsPublic && !m.IsFamily && !IsObsolete (m))
+						if (!m.IsPublic && !m.IsFamily)
+								continue;
+
+						// Availability attribute's message checks.
+						var ca = m.GetCustomAttributes<AvailabilityBaseAttribute> ();
+						if (ca.Any ()) {
+							foreach (var attr in ca) {
+								if (attr.Message != null) {
+									// Rule 1: https://github.com/xamarin/xamarin-macios/wiki/BINDINGS#rule-1
+									var forbidden = new [] { "iOS", "watchOS", "tvOS", "macOS" };
```

Comment:
@dalexsoto good comment, I improved the check so it only checks for OS strings followed by 0,1 space and a digit (for the OS version).

I checked the results of the test and they are correct, in all the cases where we mention an OS and its version number, we shouldn't because the analyzers will do that.

Things like:

```
"Never called on iOS10+" - METHOD name: GetWidgetMarginInsets, Type: NCWidgetProviding_Extensions
"Do not use; this API was removed in iOS 9.3 and will always return 0" - METHOD name: ActivityStart, Type: CKOperation
"Removed in iOS 10" - METHOD name: SetDepthClipMode, Type: MTLRenderCommandEncoder_Extensions
```